### ISSUE TITLE
Getting "readlink: illegal option -- f" when switching builds in MacOS

### ISSUE THREAD

**Issue by yaroslavvb** [tensorflow/tensorflow#2630]
I'm running into this error when building my targets with regular config, and after that trying to build with `config=cuda` in the same directory on latest MacOS (10.11.5) and bazel (0.2.3-homebrew). A work-around is to git-clone into a new directory, and build there with `config=cuda`
```

`/private/var/tmp/_bazel_yaroslavvb/1401dea20c857793b13cff5c293c0db6/tensorflow/'
____[8 / 88] Compiling external/protobuf/python/google/protobuf/internal/api_implementation.cc
____[8 / 165] Compiling external/re2/re2/filtered_re2.cc
____[8 / 313] Compiling external/boringssl_git/crypto/x509/x509_ext.c
____From Executing genrule //third_party/gpus/cuda:cuda_config_check:
readlink: illegal option -- f
usage: readlink [-n] [file ...]
ERROR: /Users/yaroslavvb/tfimmediate_fresh/tensorflow/third_party/gpus/cuda/BUILD:204:1: declared output 'third_party/gpus/cuda/cuda.config' was not created by genrule. This is probably because the genrule actually didn't create this output, or because the output was a directory and the genrule was run remotely (note that only the contents of declared file outputs are copied from genrules run remotely).
ERROR: /Users/yaroslavvb/tfimmediate_fresh/tensorflow/third_party/gpus/cuda/BUILD:204:1: not all outputs were created.
____[195 / 385] Compiling external/protobuf/src/google/protobuf/wrappers.pb.cc [for host]

```

**Comment by aselle** [comment]
Hm, yes -f doesn't exist on the Mac OS readlink.  We should probably emulate the canonicalization this way...
http://stackoverflow.com/questions/1055671/how-can-i-get-the-behavior-of-gnus-readlink-f-on-a-mac
third_party/gpus/cuda/BUILD:181

**Comment by yaroslavvb** [comment]
A side-effect of this is that if you make a mistake during "./configure" step, you must start over in a new directory. IE, running configure to change capability to "3.0" after setting it to default at first, doesn't have the desired effect, and "bazel clean" doesn't work because of this error

**Comment by gunan** [comment]
@aselle any updates here?

**Comment by rxwei** [comment]
On macOS you can use `greadlink` from homebrew'ed coreutils, which supports `-f`.

**Comment by gunan** [comment]
@yifeif has a fix for this. Yifei, could you close this issue if your PR is merged?

**Comment by yaroslavvb** [comment]
@yifeif you can have GitHub automatically reference commit/close issue if you [include](https://help.github.com/articles/closing-issues-via-commit-messages/) the following phrase anywhere in public part of CL description: closes #2630

**Comment by yifeif** [comment]
good to know @yaroslavvb :). The PR was merged a few days back. I linked this issue from it. Closing.
